### PR TITLE
Provide more information when a filewatcher failure is hit.

### PR DESCRIPTION
**Pull request by brandon-reinhart**
A naked unwrap led to an opaque error that can be hit when using the embedded filewatcher.

I've changed this an unwrap_or_else panic! with the error message providing more details about the failed operation.

A better solution would be to print an error! and not panic...

This was tested with the asset_processing example.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
This is an improvement at least. Definitely agree that it's weird that this panics.

**Comment by brandon-reinhart** [comment]
The panic may be related to what people are running into here: #10575

**Review by IceSentry — APPROVED** [review]

**Comment by brandon-reinhart** [comment]
Also #7008

File: crates/bevy_asset/src/io/file/file_watcher.rs [hunk 1]
```diff
@@ -46,7 +46,13 @@ impl FileWatcher {
 impl AssetWatcher for FileWatcher {}
 
 pub(crate) fn get_asset_path(root: &Path, absolute_path: &Path) -> (PathBuf, bool) {
-    let relative_path = absolute_path.strip_prefix(root).unwrap();
+    let relative_path = absolute_path.strip_prefix(&root).unwrap_or_else(|_| {
```
Comment (alice-i-cecile) [inline crates/bevy_asset/src/io/file/file_watcher.rs:49]:
```suggestion
    let relative_path = absolute_path.strip_prefix(root).unwrap_or_else(|_| {
```

**Pull request merged**